pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -207,6 +207,8 @@ JavaInstance setupJavaInstance() throws Exception {
 
         ContextImpl contextImpl = setupContext();
 
+        Thread.currentThread().setContextClassLoader(functionClassLoader);
```

Comment:
I don't think it is a good fix here. Because there two class loaders in the java instance, one is instance class loader and the other one is the function class loader. We set the class loader to function class loader before handling message and set it back to instance class loader.

I think a better fix is to pass the function class loader to the code that load the customized Serde.

Besides that, we should have a unit test or an integration test to verify this works.